Slicer build error with VTK9 on Windows in debug mode

When building latest master version on Windows, using Visual Studio, in debug mode, then build fails with this error:

```
  -- Detecting CXX compiler ABI info - done
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  CMake Error at C:/Program Files/CMake/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
    Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter) (found
    suitable version "3.7", minimum required is "3.6")
  Call Stack (most recent call first):
    C:/Program Files/CMake/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
    C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython/Support.cmake:2578 (find_package_handle_standard_args)
    C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython3.cmake:348 (include)
    C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/VTK-vtk-module-find-packages.cmake:303 (find_package)
    C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/vtk-config.cmake:138 (include)
    C:/D/S4D/VTK-build/vtk-config.cmake:1 (include)
    C:/D/S4D/ITK-build/lib/cmake/ITK-5.1/Modules/ITKVtkGlue.cmake:30 (find_package)
    C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:76 (include)
    C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:31 (itk_module_load)
    C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:129 (_itk_module_config_recurse)
    C:/D/S4D/ITK-build/ITKConfig.cmake:77 (itk_module_config)
    SuperBuild.cmake:386 (find_package)
    CMakeLists.txt:38 (include)


  -- Configuring incomplete, errors occurred!
  See also "C:/D/S4D/SimpleITK-build/CMakeFiles/CMakeOutput.log".
  See also "C:/D/S4D/SimpleITK-build/CMakeFiles/CMakeError.log".
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(238
,5): error MSB8066: Custom build for 'C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-mkdir.rule;C:\D\S4
D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-download.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde629
1e1\SimpleITK-update.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-patch.rule;C:\D\S4D\CMakeFiles
\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-configure.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleI
TK-build.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-install.rule;C:\D\S4D\CMakeFiles\55773c0e5
810be009a8f38bd18c05739\SimpleITK-complete.rule;C:\D\S4D\CMakeFiles\f5993a3ec64cf58c5707908b20f354e5\SimpleITK.rule;C:\
D\S4\CMakeLists.txt' exited with code 1. [C:\D\S4D\SimpleITK.vcxproj]
```

I don't know if it is related, but SimpleITK CMake cache (c:\D\S4D\SimpleITK-build\CMakeCache.txt) picks up pieces of anaconda, which is strange because my environment variables are clean (there is absolutely no mention of anaconda or python in any of the variables):

```
//Path to a program.
_Python3_CONFIG:INTERNAL=_Python3_CONFIG-NOTFOUND
_Python3_DEVELOPMENT_EMBED_SIGNATURE:INTERNAL=0993e8f7af6d2f88f03f3f71fd289925
_Python3_DEVELOPMENT_MODULE_SIGNATURE:INTERNAL=0993e8f7af6d2f88f03f3f71fd289925
//Path to a program.
_Python3_EXECUTABLE:INTERNAL=_Python3_EXECUTABLE-NOTFOUND
//Path to a file.
_Python3_INCLUDE_DIR:INTERNAL=C:/Users/andra/anaconda3/include
//Path to a library.
_Python3_LIBRARY_DEBUG:INTERNAL=_Python3_LIBRARY_DEBUG-NOTFOUND
//Path to a library.
_Python3_LIBRARY_RELEASE:INTERNAL=C:/Users/andra/anaconda3/libs/python37.lib
//Path to a library.
_Python3_RUNTIME_LIBRARY_RELEASE:INTERNAL=C:/Users/andra/anaconda3/python37.dll
//Result of TRY_COMPILE
project_compiles:INTERNAL=FALSE
```

You can find the full logs here: https://1drv.ms/u/s!Arm_AFxB9yqHxaN2aILYudYPjN5xdA?e=Fhpgcf

I've tried to build from scratch using CMake-3.19.6 - failed the same way as before (see logs in computerA-builderror-CMake-3.19.6 folder).

After I modified FindPython3 as you suggested above, I got this build error:
```
18>------ Build started: Project: SimpleITK, Configuration: Debug x64 ------
18>Performing configure step for 'SimpleITK'
18>loading initial cache file C:/D/S4D/SimpleITK-prefix/tmp/SimpleITK-cache-Debug.cmake
18>-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19042.
18>CMake Error at C:/Program Files/CMake/share/cmake-3.19/Modules/FindPython3.cmake:1 (message):
18>  Stop
18>Call Stack (most recent call first):
18>  C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/VTK-vtk-module-find-packages.cmake:303 (find_package)
18>  C:/D/S4D/VTK-build/lib/cmake/vtk-9.0/vtk-config.cmake:138 (include)
18>  C:/D/S4D/VTK-build/vtk-config.cmake:1 (include)
18>  C:/D/S4D/ITK-build/lib/cmake/ITK-5.1/Modules/ITKVtkGlue.cmake:30 (find_package)
18>  C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:76 (include)
18>  C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:31 (itk_module_load)
18>  C:/D/S4D/ITK/CMake/ITKModuleAPI.cmake:129 (_itk_module_config_recurse)
18>  C:/D/S4D/ITK-build/ITKConfig.cmake:77 (itk_module_config)
18>  SuperBuild.cmake:386 (find_package)
18>  CMakeLists.txt:38 (include)
18>
18>
18>-- Configuring incomplete, errors occurred!
18>See also "C:/D/S4D/SimpleITK-build/CMakeFiles/CMakeOutput.log".
18>See also "C:/D/S4D/SimpleITK-build/CMakeFiles/CMakeError.log".
18>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(238,5): error MSB8066: Custom build for 'C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-configure.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-build.rule;C:\D\S4D\CMakeFiles\76a3bb6e84cfb300f8ba64bbde6291e1\SimpleITK-install.rule;C:\D\S4D\CMakeFiles\55773c0e5810be009a8f38bd18c05739\SimpleITK-complete.rule;C:\D\S4D\CMakeFiles\f5993a3ec64cf58c5707908b20f354e5\SimpleITK.rule' exited with code 1.
18>Done building project "SimpleITK.vcxproj" -- FAILED.
```

And this is the CMakeCache.txt of SimpleITK:

```
# This is the CMakeCache file.
# For build in directory: c:/D/S4D/SimpleITK-build
# It was generated by CMake: C:/Program Files/CMake/bin/cmake.exe
# You can edit this file to change values found and used by cmake.
# If you do not want to change any of the values, simply exit the editor.
# If you do want to change a value, simply edit, save, and exit the editor.
# The syntax for the file is as follows:
# KEY:TYPE=VALUE
# KEY is the name of a variable in the cache.
# TYPE is a hint to GUIs for the type of VALUE, DO NOT EDIT TYPE!.
# VALUE is the current value for the KEY.

########################
# EXTERNAL cache entries
########################

//Initial cache
BUILD_DOXYGEN:BOOL=OFF

//Enable Building of the SimpleITK Examples as a separate project.
BUILD_EXAMPLES:BOOL=OFF

//Build SimpleITK ITK with shared libraries. This does not effect
// wrapped languages.
BUILD_SHARED_LIBS:BOOL=ON

//Turn on Testing for SimpleITK
BUILD_TESTING:BOOL=OFF

//Path to a program.
CMAKE_AR:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/lib.exe

//Initial cache
CMAKE_CONFIGURATION_TYPES:STRING=Debug;Release;MinSizeRel;RelWithDebInfo

//Initial cache
CMAKE_CXX_COMPILER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe

//Initial cache
CMAKE_CXX_EXTENSIONS:BOOL=OFF

//Initial cache
CMAKE_CXX_FLAGS:STRING='  /DWIN32 /D_WINDOWS /W2 /GR /EHsc '

//Flags used by the CXX compiler during DEBUG builds.
CMAKE_CXX_FLAGS_DEBUG:STRING=/MDd /Zi /Ob0 /Od /RTC1

//Flags used by the CXX compiler during MINSIZEREL builds.
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=/MD /O1 /Ob1 /DNDEBUG

//Flags used by the CXX compiler during RELEASE builds.
CMAKE_CXX_FLAGS_RELEASE:STRING=/MD /O2 /Ob2 /DNDEBUG

//Flags used by the CXX compiler during RELWITHDEBINFO builds.
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=/MD /Zi /O2 /Ob1 /DNDEBUG

//Initial cache
CMAKE_CXX_STANDARD:STRING=11

//Libraries linked by default with all C++ applications.
CMAKE_CXX_STANDARD_LIBRARIES:STRING=kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib

//Initial cache
CMAKE_CXX_STANDARD_REQUIRED:BOOL=ON

//Initial cache
CMAKE_CXX_VISIBILITY_PRESET:BOOL=default

//Initial cache
CMAKE_C_COMPILER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe

//Initial cache
CMAKE_C_FLAGS:STRING='  /DWIN32 /D_WINDOWS /W2 '

//Flags used by the C compiler during DEBUG builds.
CMAKE_C_FLAGS_DEBUG:STRING=/MDd /Zi /Ob0 /Od /RTC1

//Flags used by the C compiler during MINSIZEREL builds.
CMAKE_C_FLAGS_MINSIZEREL:STRING=/MD /O1 /Ob1 /DNDEBUG

//Flags used by the C compiler during RELEASE builds.
CMAKE_C_FLAGS_RELEASE:STRING=/MD /O2 /Ob2 /DNDEBUG

//Flags used by the C compiler during RELWITHDEBINFO builds.
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=/MD /Zi /O2 /Ob1 /DNDEBUG

//Libraries linked by default with all C applications.
CMAKE_C_STANDARD_LIBRARIES:STRING=kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib

//Flags used by the linker during all build types.
CMAKE_EXE_LINKER_FLAGS:STRING=/machine:x64

//Flags used by the linker during DEBUG builds.
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL

//Flags used by the linker during MINSIZEREL builds.
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO

//Flags used by the linker during RELEASE builds.
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO

//Flags used by the linker during RELWITHDEBINFO builds.
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL

//Initial cache
CMAKE_INSTALL_PREFIX:PATH=C:/D/S4D/SimpleITK-install

//Path to a program.
CMAKE_LINKER:FILEPATH=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/link.exe

//Flags used by the linker during the creation of modules during
// all build types.
CMAKE_MODULE_LINKER_FLAGS:STRING=/machine:x64

//Flags used by the linker during the creation of modules during
// DEBUG builds.
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL

//Flags used by the linker during the creation of modules during
// MINSIZEREL builds.
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO

//Flags used by the linker during the creation of modules during
// RELEASE builds.
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO

//Flags used by the linker during the creation of modules during
// RELWITHDEBINFO builds.
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL

//Path to a program.
CMAKE_MT:FILEPATH=C:/Program Files (x86)/Windows Kits/10/bin/10.0.18362.0/x64/mt.exe

//Value Computed by CMake
CMAKE_PROJECT_DESCRIPTION:STATIC=

//Value Computed by CMake
CMAKE_PROJECT_HOMEPAGE_URL:STATIC=

//Value Computed by CMake
CMAKE_PROJECT_NAME:STATIC=SuperBuildSimpleITK

//RC compiler
CMAKE_RC_COMPILER:FILEPATH=C:/Program Files (x86)/Windows Kits/10/bin/10.0.18362.0/x64/rc.exe

//Flags for Windows Resource Compiler during all build types.
CMAKE_RC_FLAGS:STRING=-DWIN32

//Flags for Windows Resource Compiler during DEBUG builds.
CMAKE_RC_FLAGS_DEBUG:STRING=-D_DEBUG

//Flags for Windows Resource Compiler during MINSIZEREL builds.
CMAKE_RC_FLAGS_MINSIZEREL:STRING=

//Flags for Windows Resource Compiler during RELEASE builds.
CMAKE_RC_FLAGS_RELEASE:STRING=

//Flags for Windows Resource Compiler during RELWITHDEBINFO builds.
CMAKE_RC_FLAGS_RELWITHDEBINFO:STRING=

//Flags used by the linker during the creation of shared libraries
// during all build types.
CMAKE_SHARED_LINKER_FLAGS:STRING=/machine:x64

//Flags used by the linker during the creation of shared libraries
// during DEBUG builds.
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=/debug /INCREMENTAL

//Flags used by the linker during the creation of shared libraries
// during MINSIZEREL builds.
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=/INCREMENTAL:NO

//Flags used by the linker during the creation of shared libraries
// during RELEASE builds.
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=/INCREMENTAL:NO

//Flags used by the linker during the creation of shared libraries
// during RELWITHDEBINFO builds.
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=/debug /INCREMENTAL

//If set, runtime paths are not added when installing shared libraries,
// but are added when building.
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO

//If set, runtime paths are not added when using shared libraries.
CMAKE_SKIP_RPATH:BOOL=NO

//Flags used by the linker during the creation of static libraries
// during all build types.
CMAKE_STATIC_LINKER_FLAGS:STRING=/machine:x64

//Flags used by the linker during the creation of static libraries
// during DEBUG builds.
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=

//Flags used by the linker during the creation of static libraries
// during MINSIZEREL builds.
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=

//Flags used by the linker during the creation of static libraries
// during RELEASE builds.
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=

//Flags used by the linker during the creation of static libraries
// during RELWITHDEBINFO builds.
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=

//If this value is on, makefiles will be generated without the
// .SILENT directive, and all commands will be echoed to the console
// during the make.  This is useful for debugging only. With Visual
// Studio IDE projects all commands are done without /nologo.
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE

//linker flags for dynamic lookup (when linking MODULE against
// SHARED)
DYNAMIC_LOOKUP_FLAGS_MODULE_SHARED:STRING=

//Initial cache
ExternalData_OBJECT_STORES:PATH=C:/D/S4D/ExternalData/Objects

//Additional URL templates for the ExternalData CMake script to
// look for testing data. E.g.
//\nfile:///var/bigharddrive/%(algo)/%(hash)
ExternalData_URL_TEMPLATES:STRING=

//Git command line client
GIT_EXECUTABLE:FILEPATH=C:/Program Files/Git/cmd/git.exe

//linker supports dynamic lookup for undefined symbols (when linking
// MODULE against SHARED)
HAS_DYNAMIC_LOOKUP_MODULE_SHARED:BOOL=FALSE

//Initial cache
ITK_DIR:PATH=C:/D/S4D/ITK-build

//OpenGL library for win32
OPENGL_gl_LIBRARY:STRING=opengl32

//GLU library for win32
OPENGL_glu_LIBRARY:STRING=glu32

//Path to a library.
PYTHON_DEBUG_LIBRARY:FILEPATH=PYTHON_DEBUG_LIBRARY-NOTFOUND

//Initial cache
PYTHON_EXECUTABLE:PATH=C:/D/S4D/python-install/bin/PythonSlicer.exe

//Initial cache
PYTHON_INCLUDE_DIR:PATH=C:/D/S4D/python-install/include

//Initial cache
PYTHON_LIBRARY:FILEPATH=C:/D/S4D/python-install/libs/python36.lib

//Path to a library.
PYTHON_LIBRARY_DEBUG:FILEPATH=PYTHON_LIBRARY_DEBUG-NOTFOUND

//The directory containing a CMake configuration file for Qt5Core.
Qt5Core_DIR:PATH=C:/Qt/5.15.0/msvc2019_64/lib/cmake/Qt5Core

//The directory containing a CMake configuration file for Qt5Gui.
Qt5Gui_DIR:PATH=C:/Qt/5.15.0/msvc2019_64/lib/cmake/Qt5Gui

//The directory containing a CMake configuration file for Qt5Widgets.
Qt5Widgets_DIR:PATH=C:/Qt/5.15.0/msvc2019_64/lib/cmake/Qt5Widgets

//Third-party helper setting from VTK
Qt5_DIR:FILEPATH=C:/Qt/5.15.0/msvc2019_64/lib/cmake/Qt5

//Path to a file.
SWIG_DIR:PATH=C:/D/S4D/swigwin-4.0.2/Lib

//Initial cache
SWIG_EXECUTABLE:PATH=C:/D/S4D/swigwin-4.0.2/swig.exe

//Swig version
SWIG_VERSION:STRING=4.0.2

//Initial cache
SimpleITK_BUILD_DISTRIBUTE:BOOL=ON

//Initial cache
SimpleITK_BUILD_STRIP:BOOL=1

//If behind a firewall turn set this to 'https' or 'http'.
SimpleITK_GIT_PROTOCOL:STRING=https

//Initial cache
SimpleITK_INSTALL_ARCHIVE_DIR:PATH=./lib/Slicer-4.13

//Initial cache
SimpleITK_INSTALL_LIBRARY_DIR:PATH=./lib/Slicer-4.13

//Initial cache
SimpleITK_INT64_PIXELIDS:BOOL=OFF

//Add Image and I/O support up to max dimension (integer).
SimpleITK_MAX_DIMENSION:STRING=5

//Initial cache
SimpleITK_PYTHON_THREADS:BOOL=ON

//Use an external version of GoogleTest. 
SimpleITK_USE_SYSTEM_GTEST:BOOL=OFF

//Use a pre-built version of ITK
SimpleITK_USE_SYSTEM_ITK:BOOL=ON

//Use a pre-compiled version of LUA 5.3 previously configured for
// your system
SimpleITK_USE_SYSTEM_LUA:BOOL=OFF

//Use a pre-compiled version of SWIG 3.0 previously configured
// for your system
SimpleITK_USE_SYSTEM_SWIG:BOOL=ON

//Use a system version of Python's virtualenv. 
SimpleITK_USE_SYSTEM_VIRTUALENV:BOOL=OFF

//Value Computed by CMake
SuperBuildSimpleITK_BINARY_DIR:STATIC=C:/D/S4D/SimpleITK-build

//Value Computed by CMake
SuperBuildSimpleITK_SOURCE_DIR:STATIC=C:/D/S4D/SimpleITK/SuperBuild

//Use ccache to expedite re-compilations. Recommended for development
USE_CCACHE:BOOL=OFF

//Use ld.gold linker (decreases linking time if available)
USE_GOLD_LINKER:BOOL=OFF

//Wrap C#
WRAP_CSHARP:BOOL=OFF

//Enables automatic detection and wrapping for all languages.
WRAP_DEFAULT:BOOL=OFF

//Wrap Java
WRAP_JAVA:BOOL=OFF

//Wrap Lua
WRAP_LUA:BOOL=OFF

//Wrap Python
WRAP_PYTHON:BOOL=ON

//Wrap R
WRAP_R:BOOL=OFF

//Wrap Ruby
WRAP_RUBY:BOOL=OFF

//Wrap Tcl
WRAP_TCL:BOOL=OFF


########################
# INTERNAL cache entries
########################

//ADVANCED property for variable: BUILD_SHARED_LIBS
BUILD_SHARED_LIBS-ADVANCED:INTERNAL=1
//This is the directory where this CMakeCache.txt was created
CMAKE_CACHEFILE_DIR:INTERNAL=c:/D/S4D/SimpleITK-build
//Major version of cmake used to create the current loaded cache
CMAKE_CACHE_MAJOR_VERSION:INTERNAL=3
//Minor version of cmake used to create the current loaded cache
CMAKE_CACHE_MINOR_VERSION:INTERNAL=19
//Patch version of cmake used to create the current loaded cache
CMAKE_CACHE_PATCH_VERSION:INTERNAL=6
//Path to CMake executable.
CMAKE_COMMAND:INTERNAL=C:/Program Files/CMake/bin/cmake.exe
//Path to cpack program executable.
CMAKE_CPACK_COMMAND:INTERNAL=C:/Program Files/CMake/bin/cpack.exe
//Path to ctest program executable.
CMAKE_CTEST_COMMAND:INTERNAL=C:/Program Files/CMake/bin/ctest.exe
//ADVANCED property for variable: CMAKE_CXX_FLAGS
CMAKE_CXX_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_CXX_FLAGS_DEBUG
CMAKE_CXX_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_CXX_FLAGS_MINSIZEREL
CMAKE_CXX_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_CXX_FLAGS_RELEASE
CMAKE_CXX_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_CXX_FLAGS_RELWITHDEBINFO
CMAKE_CXX_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_CXX_STANDARD_LIBRARIES
CMAKE_CXX_STANDARD_LIBRARIES-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_FLAGS
CMAKE_C_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_FLAGS_DEBUG
CMAKE_C_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_FLAGS_MINSIZEREL
CMAKE_C_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_FLAGS_RELEASE
CMAKE_C_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_FLAGS_RELWITHDEBINFO
CMAKE_C_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_C_STANDARD_LIBRARIES
CMAKE_C_STANDARD_LIBRARIES-ADVANCED:INTERNAL=1
//Executable file format
CMAKE_EXECUTABLE_FORMAT:INTERNAL=Unknown
//ADVANCED property for variable: CMAKE_EXE_LINKER_FLAGS
CMAKE_EXE_LINKER_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_EXE_LINKER_FLAGS_DEBUG
CMAKE_EXE_LINKER_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_EXE_LINKER_FLAGS_MINSIZEREL
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_EXE_LINKER_FLAGS_RELEASE
CMAKE_EXE_LINKER_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//Name of external makefile project generator.
CMAKE_EXTRA_GENERATOR:INTERNAL=
//Name of generator.
CMAKE_GENERATOR:INTERNAL=Visual Studio 16 2019
//Generator instance identifier.
CMAKE_GENERATOR_INSTANCE:INTERNAL=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community
//Name of generator platform.
CMAKE_GENERATOR_PLATFORM:INTERNAL=
//Name of generator toolset.
CMAKE_GENERATOR_TOOLSET:INTERNAL=v142
//Have include pthread.h
CMAKE_HAVE_PTHREAD_H:INTERNAL=
//Source directory with the top level CMakeLists.txt file for this
// project
CMAKE_HOME_DIRECTORY:INTERNAL=C:/D/S4D/SimpleITK/SuperBuild
//ADVANCED property for variable: CMAKE_LINKER
CMAKE_LINKER-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MODULE_LINKER_FLAGS
CMAKE_MODULE_LINKER_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MODULE_LINKER_FLAGS_DEBUG
CMAKE_MODULE_LINKER_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MODULE_LINKER_FLAGS_RELEASE
CMAKE_MODULE_LINKER_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_MT
CMAKE_MT-ADVANCED:INTERNAL=1
//number of local generators
CMAKE_NUMBER_OF_MAKEFILES:INTERNAL=1
//Platform information initialized
CMAKE_PLATFORM_INFO_INITIALIZED:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_COMPILER
CMAKE_RC_COMPILER-ADVANCED:INTERNAL=1
CMAKE_RC_COMPILER_WORKS:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_FLAGS
CMAKE_RC_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_FLAGS_DEBUG
CMAKE_RC_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_FLAGS_MINSIZEREL
CMAKE_RC_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_FLAGS_RELEASE
CMAKE_RC_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_RC_FLAGS_RELWITHDEBINFO
CMAKE_RC_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//Path to CMake installation.
CMAKE_ROOT:INTERNAL=C:/Program Files/CMake/share/cmake-3.19
//ADVANCED property for variable: CMAKE_SHARED_LINKER_FLAGS
CMAKE_SHARED_LINKER_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SHARED_LINKER_FLAGS_DEBUG
CMAKE_SHARED_LINKER_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SHARED_LINKER_FLAGS_RELEASE
CMAKE_SHARED_LINKER_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SKIP_INSTALL_RPATH
CMAKE_SKIP_INSTALL_RPATH-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_SKIP_RPATH
CMAKE_SKIP_RPATH-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_STATIC_LINKER_FLAGS
CMAKE_STATIC_LINKER_FLAGS-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_STATIC_LINKER_FLAGS_DEBUG
CMAKE_STATIC_LINKER_FLAGS_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_STATIC_LINKER_FLAGS_RELEASE
CMAKE_STATIC_LINKER_FLAGS_RELEASE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=1
//ADVANCED property for variable: CMAKE_VERBOSE_MAKEFILE
CMAKE_VERBOSE_MAKEFILE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: DYNAMIC_LOOKUP_FLAGS_MODULE_SHARED
DYNAMIC_LOOKUP_FLAGS_MODULE_SHARED-ADVANCED:INTERNAL=1
//ADVANCED property for variable: ExternalData_OBJECT_STORES
ExternalData_OBJECT_STORES-ADVANCED:INTERNAL=1
//ADVANCED property for variable: ExternalData_URL_TEMPLATES
ExternalData_URL_TEMPLATES-ADVANCED:INTERNAL=1
//Details about finding Git
FIND_PACKAGE_MESSAGE_DETAILS_Git:INTERNAL=[C:/Program Files/Git/cmd/git.exe][v2.28.0.windows.1()]
//Details about finding OpenGL
FIND_PACKAGE_MESSAGE_DETAILS_OpenGL:INTERNAL=[opengl32][cfound components: OpenGL ][v()]
//Details about finding PythonInterp
FIND_PACKAGE_MESSAGE_DETAILS_PythonInterp:INTERNAL=[C:/D/S4D/python-install/bin/PythonSlicer.exe][v3.6.7()]
//Details about finding PythonLibs
FIND_PACKAGE_MESSAGE_DETAILS_PythonLibs:INTERNAL=[C:/D/S4D/python-install/libs/python36.lib][C:/D/S4D/python-install/include][v3.6.7(3.6.7)]
//Details about finding SWIG
FIND_PACKAGE_MESSAGE_DETAILS_SWIG:INTERNAL=[C:/D/S4D/swigwin-4.0.2/swig.exe][C:/D/S4D/swigwin-4.0.2/Lib][c ][v4.0.2(3)]
//Details about finding Threads
FIND_PACKAGE_MESSAGE_DETAILS_Threads:INTERNAL=[TRUE][v()]
//ADVANCED property for variable: GIT_EXECUTABLE
GIT_EXECUTABLE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: HAS_DYNAMIC_LOOKUP_MODULE_SHARED
HAS_DYNAMIC_LOOKUP_MODULE_SHARED-ADVANCED:INTERNAL=1
//hashed flags for HAS_DYNAMIC_LOOKUP_MODULE_SHARED check
HAS_DYNAMIC_LOOKUP_MODULE_SHARED_hash:INTERNAL=e0acce47614caaf9b71292c58d7cd8b2
//ADVANCED property for variable: OPENGL_gl_LIBRARY
OPENGL_gl_LIBRARY-ADVANCED:INTERNAL=1
//ADVANCED property for variable: OPENGL_glu_LIBRARY
OPENGL_glu_LIBRARY-ADVANCED:INTERNAL=1
//ADVANCED property for variable: PYTHON_DEBUG_LIBRARY
PYTHON_DEBUG_LIBRARY-ADVANCED:INTERNAL=1
//ADVANCED property for variable: PYTHON_EXECUTABLE
PYTHON_EXECUTABLE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: PYTHON_INCLUDE_DIR
PYTHON_INCLUDE_DIR-ADVANCED:INTERNAL=1
//ADVANCED property for variable: PYTHON_LIBRARY
PYTHON_LIBRARY-ADVANCED:INTERNAL=1
//ADVANCED property for variable: PYTHON_LIBRARY_DEBUG
PYTHON_LIBRARY_DEBUG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SWIG_DIR
SWIG_DIR-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SWIG_EXECUTABLE
SWIG_EXECUTABLE-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SWIG_VERSION
SWIG_VERSION-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_GIT_PROTOCOL
SimpleITK_GIT_PROTOCOL-ADVANCED:INTERNAL=1
//STRINGS property for variable: SimpleITK_GIT_PROTOCOL
SimpleITK_GIT_PROTOCOL-STRINGS:INTERNAL=https;http;git
//ADVANCED property for variable: SimpleITK_MAX_DIMENSION
SimpleITK_MAX_DIMENSION-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_USE_SYSTEM_GTEST
SimpleITK_USE_SYSTEM_GTEST-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_USE_SYSTEM_ITK
SimpleITK_USE_SYSTEM_ITK-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_USE_SYSTEM_LUA
SimpleITK_USE_SYSTEM_LUA-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_USE_SYSTEM_SWIG
SimpleITK_USE_SYSTEM_SWIG-ADVANCED:INTERNAL=1
//ADVANCED property for variable: SimpleITK_USE_SYSTEM_VIRTUALENV
SimpleITK_USE_SYSTEM_VIRTUALENV-ADVANCED:INTERNAL=1
//ADVANCED property for variable: USE_GOLD_LINKER
USE_GOLD_LINKER-ADVANCED:INTERNAL=1
//ADVANCED property for variable: WRAP_DEFAULT
WRAP_DEFAULT-ADVANCED:INTERNAL=1
//ADVANCED property for variable: WRAP_R
WRAP_R-ADVANCED:INTERNAL=1
//Path to a program.
_Python3_CONFIG:INTERNAL=_Python3_CONFIG-NOTFOUND
_Python3_DEVELOPMENT_EMBED_SIGNATURE:INTERNAL=4e66d214c2a025db51c924cf55f2343e
_Python3_DEVELOPMENT_MODULE_SIGNATURE:INTERNAL=4e66d214c2a025db51c924cf55f2343e
//Path to a program.
_Python3_EXECUTABLE:INTERNAL=_Python3_EXECUTABLE-NOTFOUND
//Path to a file.
_Python3_INCLUDE_DIR:INTERNAL=C:/Users/andra/anaconda3/include
//Path to a library.
_Python3_LIBRARY_DEBUG:INTERNAL=_Python3_LIBRARY_DEBUG-NOTFOUND
//Path to a library.
_Python3_LIBRARY_RELEASE:INTERNAL=C:/Users/andra/anaconda3/libs/python37.lib
//Path to a library.
_Python3_RUNTIME_LIBRARY_RELEASE:INTERNAL=C:/Users/andra/anaconda3/python37.dll
//Result of TRY_COMPILE
project_compiles:INTERNAL=FALSE
```

I get a similar error in Ubuntu 20.10 (for both cmake 3.16.3 and cmake 3.19.6):

```
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find Python3 (missing: Python3_INCLUDE_DIRS Development.Module)
  (found suitable version "3.8.6", minimum required is "3.6")
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython/Support.cmake:2578 (find_package_handle_standard_args)
  /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython3.cmake:348 (include)
  /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/VTK-vtk-module-find-packages.cmake:350 (find_package)
  /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/vtk-config.cmake:138 (include)
  /home/rafael/src/Slicer-vtk9/build/VTK-build/vtk-config.cmake:1 (include)
  /home/rafael/src/Slicer-vtk9/build/ITK-build/lib/cmake/ITK-5.1/Modules/ITKVtkGlue.cmake:30 (find_package)
  /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:76 (include)
  /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:31 (itk_module_load)
  /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:129 (_itk_module_config_recurse)
  /home/rafael/src/Slicer-vtk9/build/ITK-build/ITKConfig.cmake:77 (itk_module_config)
  SuperBuild.cmake:386 (find_package)
  CMakeLists.txt:38 (include)
```

I have tried the changes @lassoan suggested but the problem persists. I have also tried to replace `-DPython3_INCLUDE_DIR` by `-DPython3_INCLUDE_DIRS` (the error uses _DIR**S**); still no luck.

> I get a similar error in Ubuntu 20.10 (for both cmake 3.16.3 and cmake 3.19.6):
> 
> ```
> CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
>   Could NOT find Python3 (missing: Python3_INCLUDE_DIRS Development.Module)
>   (found suitable version "3.8.6", minimum required is "3.6")
> Call Stack (most recent call first):
>   /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
>   /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython/Support.cmake:2578 (find_package_handle_standard_args)
>   /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/patches/3.18/FindPython3.cmake:348 (include)
>   /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/VTK-vtk-module-find-packages.cmake:350 (find_package)
>   /home/rafael/src/Slicer-vtk9/build/VTK-build/lib/cmake/vtk-9.0/vtk-config.cmake:138 (include)
>   /home/rafael/src/Slicer-vtk9/build/VTK-build/vtk-config.cmake:1 (include)
>   /home/rafael/src/Slicer-vtk9/build/ITK-build/lib/cmake/ITK-5.1/Modules/ITKVtkGlue.cmake:30 (find_package)
>   /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:76 (include)
>   /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:31 (itk_module_load)
>   /home/rafael/src/Slicer-vtk9/build/ITK/CMake/ITKModuleAPI.cmake:129 (_itk_module_config_recurse)
>   /home/rafael/src/Slicer-vtk9/build/ITK-build/ITKConfig.cmake:77 (itk_module_config)
>   SuperBuild.cmake:386 (find_package)
>   CMakeLists.txt:38 (include)
> ```

This can now be reproduced with dockbuild/dockbuild#73 and [Slicer/SlicerBuildEnvironment](https://github.com/Slicer/SlicerBuildEnvironment)

--- a/SuperBuild/External_VTK.cmake
+++ b/SuperBuild/External_VTK.cmake
@@ -177,7 +177,7 @@ if((NOT DEFINED VTK_DIR OR NOT DEFINED VTK_SOURCE_DIR) AND NOT Slicer_USE_SYSTEM
     set(_git_tag "97904fdcc7e73446b3131f32eac9fc9781b23c2d") # slicer-v8.2.0-2018-10-02-74d9488523
     set(vtk_egg_info_version "8.2.0")
   elseif("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "9")
-    set(_git_tag "558ee1c5be28853cf2316bc3dcc75dc66bf81243") # slicer-v9.0.20201111-733234c785-3
+    set(_git_tag "558ee1c5be28853cf2316bc3dcc75dc66bf81243") # slicer-v9.0.20201111-733234c785-v3
     set(vtk_egg_info_version "9.0.20201111")
   else()
     message(FATAL_ERROR "error: Unsupported Slicer_VTK_VERSION_MAJOR: ${Slicer_VTK_VERSION_MAJOR}")

